service/src/test/java/org/dice_research/fc/paths/scorer/count/max/MaxCounterTest.java

```diff
@@ -0,0 +1,74 @@
+package org.dice_research.fc.paths.scorer.count.max;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.aksw.jena_sparql_api.core.QueryExecutionFactory;
+import org.aksw.jena_sparql_api.http.QueryExecutionFactoryHttp;
+import org.apache.jena.rdf.model.Property;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.rdf.model.Statement;
+import org.dice_research.fc.data.QRestrictedPath;
+import org.dice_research.fc.paths.EmptyPredicateFactory;
+import org.dice_research.fc.paths.FactPreprocessor;
+import org.dice_research.fc.paths.PredicateFactory;
+import org.dice_research.fc.paths.scorer.ICountRetriever;
+import org.dice_research.fc.paths.scorer.NPMIBasedScorer;
+import org.dice_research.fc.paths.scorer.count.ApproximatingCountRetriever;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Testing <http://dbpedia.org/ontology/birthPlace>/<http://dbpedia.org/ontology/country> with and
+ * without virtual type configuration.
+ * 
+ * @author Alexandra Silva
+ *
+ */
+@RunWith(Parameterized.class)
+public class MaxCounterTest {
+  private Statement fact = ResourceFactory.createStatement(
+      ResourceFactory.createResource("http://dbpedia.org/resource/Bill_Gates"),
+      ResourceFactory.createProperty("http://dbpedia.org/ontology/birthPlace"),
+      ResourceFactory.createResource("http://dbpedia.org/resource/United_States"));
+
+  private Property[] properties =
+      {ResourceFactory.createProperty("http://dbpedia.org/ontology/birthPlace"),
+          ResourceFactory.createProperty("http://dbpedia.org/ontology/country")};
+  private boolean[] directions = {true, true};
+
+  private static QueryExecutionFactory qef =
+      new QueryExecutionFactoryHttp("https://synthg-fact.dice-research.org/sparql");
```

Comment:
Having a test is really great and we definitely should have one. However, it is not a good idea to rely on something that is "external" to the project.

I think it would be much easier and saver to create some simple local example graphs. In addition, we should always know exactly what we expect. Checking that the result is not empty is not bad, but it does not give us a lot of safety.

For the virtual types, the test examples could be:
1. `a p b . a p c .` expected result: 2
2. `a p b . a p c . d p c .` expected result: 4
3. `a q b .` expected result: 0

Similar examples for the default implementation are straight forward :wink: